nexus-sled-agent-shared/src/inventory.rs

```diff
@@ -131,6 +128,50 @@ pub struct ConfigReconcilerInventory {
     pub zones: BTreeMap<OmicronZoneUuid, ConfigReconcilerInventoryResult>,
 }
 
+impl ConfigReconcilerInventory {
+    /// Iterate over all running zones as reported by the last reconciliation
+    /// result.
+    ///
+    /// This includes zones that are both present in `last_reconciled_config`
+    /// and whose status in `zones` indicates "successfully running".
+    pub fn running_omicron_zones(
+        &self,
+    ) -> impl Iterator<Item = &OmicronZoneConfig> {
+        self.zones.iter().filter_map(|(zone_id, result)| {
+            match result {
+                ConfigReconcilerInventoryResult::Ok => (),
+                ConfigReconcilerInventoryResult::Err { .. } => return None,
+            };
+            self.last_reconciled_config.zones.get(zone_id)
+        })
+    }
+
+    /// Given a sled config, produce a reconciler result that sled-agent could
+    /// have emitted if reconciliation succeeded.
+    ///
+    /// This method should only be used by tests and dev tools; real code should
```

Comment:
Should we mark this with `#[test]` and maybe `#[cfg(any(test, feature = "testing"))]` ?

Reply:
I don't think we can, because of the "and dev tools". IIRC reconfigurator-cli and some of the "example system" stuff uses this, neither of which is gated by `test` / `testing`.